programs/conditional_vault/src/instructions/resolve_question.rs

```diff
@@ -0,0 +1,33 @@
+use super::*;
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct ResolveQuestionArgs {
+    pub payout_numerators: Vec<u32>,
+}
+
+#[derive(Accounts)]
+#[instruction(args: ResolveQuestionArgs)]
+pub struct ResolveQuestion<'info> {
+    #[account(mut, has_one = oracle)]
+    pub question: Account<'info, Question>,
+    pub oracle: Signer<'info>,
+}
+
+impl ResolveQuestion<'_> {
+    pub fn handle(ctx: Context<Self>, args: ResolveQuestionArgs) -> Result<()> {
+        let question = &mut ctx.accounts.question;
+
+        require_eq!(
+            args.payout_numerators.len(),
+            question.num_outcomes(),
+            VaultError::InvalidNumPayoutNumerators
+        );
+
```

Comment:
I think we require a check that there is a winning outcome. 

// Find the maximum value
let max_value = *args.payout_numerators.iter().max().unwrap_or(&0);

// Check if the maximum value is unique
if args.payout_numerators.iter().filter(|&&x| x == max_value).count() != 1 {
    return Err(VaultError::NonUniquePayout);
}